TaskWithSmartPointers/MySharedPtr.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+#include <iostream>
+
+template <class T>
+class MySharedPtr
+{
+private:
```

Comment:
The good manner is to declare private members at the end of the class definitions, after public and protected members (if any).

Reply:
Done